src/test/scala/cromwell/binding/values/WdlValueSpec.scala

```diff
@@ -0,0 +1,101 @@
+package cromwell.binding.values
+
+import java.nio.file.Paths
+
+import cromwell.binding.{WdlExpression, WdlNamespace}
+import cromwell.util.SampleWdl
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.{FlatSpec, Matchers}
+
+class WdlValueSpec extends FlatSpec with Matchers {
+
+  import TableDrivenPropertyChecks._
+
+  behavior of "WdlValue"
+
+  val wdlValueRawStrings = Table(
+    ("wdlValue", "rawString"),
+    (WdlBoolean(false), "false"),
+    (WdlBoolean(true), "true"),
+    (new WdlFile(Paths.get("hello/world/path")), "hello/world/path"),
+    (WdlFile("hello/world/string"), "hello/world/string"),
+    (WdlFloat(0.0), "0.0"),
+    (WdlFloat(-0.0), "-0.0"),
+    (WdlFloat(Double.PositiveInfinity), "Infinity"),
+    (WdlFloat(Double.NegativeInfinity), "-Infinity"),
+    (WdlInteger(0), "0"),
+    (WdlInteger(Int.MaxValue), "2147483647"),
+    (WdlInteger(Int.MinValue), "-2147483648"),
+    (WdlString(""), ""),
+    (WdlString("test\\'/string"), "test\\'/string"))
+
+  forAll(wdlValueRawStrings) { (wdlValue, rawString) =>
+    it should s"exactly convert a Wdl${wdlValue.wdlType.toWdlString} to/from raw string '$rawString'" in {
+      val toRawString = wdlValue.toRawString
+      toRawString should be(rawString)
+
+      val wdlType = wdlValue.wdlType
+      val fromRawString = wdlType.fromRawString(toRawString)
+      fromRawString should be(wdlValue)
+      fromRawString.wdlType should be(wdlType)
+    }
+  }
+
+  val wdlFloatTruncated = Table(
+    ("wdlValue", "rawString", "validateFloat"),
+    (WdlFloat(Double.MinPositiveValue), "4.9E-324", { d: Double => d == 0.0 }),
+    (WdlFloat(Double.NaN), "NaN", { d: Double => d.isNaN }),
+    (WdlFloat(Double.MaxValue), "1.7976931348623157E308", { d: Double => d.isPosInfinity }),
+    (WdlFloat(Double.MinValue), "-1.7976931348623157E308", { d: Double => d.isNegInfinity }))
+
+  forAll(wdlFloatTruncated) { (wdlValue, rawString, validateFloat) =>
+    it should s"truncate a Wdl${wdlValue.wdlType.toWdlString} to/from raw string '$rawString'" in {
```

Comment:
Not sure I follow why we're testing float truncation, or even how except for the first case?